whats-that-typeclass-functor/article.md

```diff
@@ -0,0 +1,467 @@
+You might already know the `map` function for lists:
+
+```haskell
+> map (+1) [1, 2, 3]
+[2, 3, 4]
+```
+
+It takes a function of type `a -> b` and applies it to each element of a list of type `[a]`. The values change, but the context (`[]`) remains the same.
+
+Transforming a value inside a fixed context like that is common in Haskell.
```

Comment:
> is common in Haskell.

But isn't this operation common in all languages? And Haskell and the likes just provides ways to generalize it?

Reply:
That's a good way to introduce that that matches with how we talk about fmap, I like that.

Reply:
> And Haskell and the likes just provides ways to generalize it?

I wouldn't say that it generalises anything, to be fair. Languages with wonkier semantics have more expressive transformers of data! (Because looser semantics means more expressivity, at a cost of correctness).

I would think of Haskell as a language that provides taxonomies for such transformers of data!

Reply:
Changed a bit. Does it look clearer and more fair now?

Reply:
I like the phrase "provides taxonomies" tbh. 

I will leave this open for now since the whole intro is a bit shaky, needs rewording/re-arranging of sentences, which I'll look at.

Reply:
The intro has been reworded, so I'll mark this as resolved now.